account-management-api/src/main/java/uk/gov/di/accountmanagement/lambda/NotificationHandler.java

```diff
@@ -58,104 +59,104 @@ public NotificationHandler(ConfigurationService configurationService) {
-                                    "contact-us-link", buildContactUsUrl());
-                            notificationService.sendEmail(
-                                    notifyRequest.getDestination(),
-                                    accountDeletedPersonalisation,
-                                    NotificationType.DELETE_ACCOUNT);
-                            LOG.info("DELETE_ACCOUNT email has been sent using Notify");
-                            break;
-                        case PHONE_NUMBER_UPDATED:
-                            LOG.info("Sending PHONE_NUMBER_UPDATED email using Notify");
-                            Map<String, Object> phoneNumberUpdatedPersonalisation = new HashMap<>();
-                            phoneNumberUpdatedPersonalisation.put(
-                                    "contact-us-link", buildContactUsUrl());
-                            notificationService.sendEmail(
-                                    notifyRequest.getDestination(),
-                                    phoneNumberUpdatedPersonalisation,
-                                    NotificationType.PHONE_NUMBER_UPDATED);
-                            LOG.info("PHONE_NUMBER_UPDATED email has been sent using Notify");
-                            break;
-                        case PASSWORD_UPDATED:
-                            LOG.info("Sending PASSWORD_UPDATED email using Notify");
-                            Map<String, Object> passwordUpdatedPersonalisation = new HashMap<>();
-                            passwordUpdatedPersonalisation.put(
-                                    "contact-us-link", buildContactUsUrl());
-                            notificationService.sendEmail(
-                                    notifyRequest.getDestination(),
-                                    passwordUpdatedPersonalisation,
-                                    NotificationType.PASSWORD_UPDATED);
-                            LOG.info("PASSWORD_UPDATED email has been sent using Notify");
-                            break;
-                    }
-                } catch (NotificationClientException e) {
-                    LOG.error("Error sending with Notify", e);
-                    throw new RuntimeException(
-                            String.format(
-                                    "Error sending with Notify using NotificationType: %s",
-                                    notifyRequest.getNotificationType()),
-                            e);
-                }
+                processNotificationRequest(notifyRequest);
             } catch (JsonException e) {
-                LOG.error("Error when mapping message from queue to a NotifyRequest");
-                throw new RuntimeException(
-                        "Error when mapping message from queue to a NotifyRequest");
+                LOG.error("Error when mapping message from queue to a NotifyRequest", e);
+                throw new NotificationClientException(
```

Comment:
Throwing an exception from the lambda will cause it to be restarted.  Rather than refactoring the exception to a new more specific type it would be better to remove the thrown exception if the error is temporary and the next request will likely be successful.